packages/website/components/MetamaskNetworkButton/ConnectToMetamask.tsx

```diff
@@ -0,0 +1,87 @@
+type Props = {
+  buttonText: string;
+};
+
+async function ConnectToMetamask(props: Props) {
+  if (!(window as any).ethereum) {
+    alert("Wallet not detected! Install Metamask then try again.");
+    return;
+  }
+  if(props.buttonText == "Connect to Sepolia"){
+    ConnectToSepolia();
+  }
+  if(props.buttonText == "Connect to Taiko"){
+    ConnectToTaiko();
+  }
+}
+
+async function ConnectToSepolia() {
+  if (!(window as any).ethereum) {
+    alert("Metamask not detected! Install Metamask then try again.")
+    return;
+  }
+  if ((window as any).ethereum.networkVersion == "11155111") {
+    alert("You are already connected to Sepolia (chainId 11155111).", )
+    return;
+  }
+  try{
+    await (window as any).ethereum.request({
+      method: "wallet_switchEthereumChain",
+      params: [{
+         chainId: "0xaa36a7"
+      }]
+    });
+  } catch (error) {
+    alert("Failed to add the network with wallet_addEthereumChain request. Add the network with https://chainlist.org/ or do it manually. Error log: " + error.message)
+  }
+}
+
+async function ConnectToTaiko() {
+  if ((window as any).ethereum.networkVersion == "167005") {
+    alert("You are already connected to Taiko Alpha 3 (chainId 167005).", )
+    return;
+  }
+  interface AddEthereumChainParameter {
+    chainId: string; // A 0x-prefixed hexadecimal string
+    chainName: string;
+    nativeCurrency: {
+      name: string;
+      symbol: string; // 2-6 characters long
+      decimals: 18;
+    };
+    rpcUrls: string[];
+    blockExplorerUrls?: string[];
+    iconUrls?: string[]; // Currently ignored.
+  }
+  const taikoParams: AddEthereumChainParameter = {
+    chainId: "0x28c5d",
+    chainName: "Taiko (Alpha-3 Testnet)",
+    nativeCurrency: {
+      name: "ETH",
+      symbol: "eth",
+      decimals: 18,
+    },
+    rpcUrls: ["https://rpc.test.taiko.xyz"],
+    blockExplorerUrls: ["https://explorer.test.taiko.xyz/"],
+    iconUrls: [],
```

Comment:
can we make sure it matches this exactly: https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-167005.json

otherwise, that alert error will through. i already added taiko from chainlist, and if i click it, it says the symbol is mismatched (i think because of eth vs ETH)

Reply:
Updated currency name and symbol.

With 
```js
chainId: "167005",
```
I see error
```js
Expected 0x-prefixed, unpadded, non-zero hexadecimal string 'chainId'. Received: 167005
```
therefore we should keep the chainId in hexadecimal format
```js
chainId: "0x28c5d",
```